lib/lcevc/lcevc_dec.js

```diff
@@ -0,0 +1,216 @@
+    /** @private {?} */
+    // LCEVC Decoder library based on availability, to check if either
+    // lcevc_dil or lcevc_dec is present.
+    this.lcevcLib_ = null;
+
+    /** @private {?LCEVCdec.LCEVCdec} */
+    this.dec_ = null;
+
+    /** @private {number} */
+    this.variantId_ = -1;
+
+    /** @private {HTMLVideoElement} */
+    this.media_ = media;
+
+    /** @private {HTMLCanvasElement} */
+    this.canvas_ = canvas;
+
+    /** @private {shaka.extern.LcevcConfiguration} */
+    this.decConfig_ = lcevcConfig;
+
+    /** @private {boolean} */
+    this.toBeDeprecated_ = false;
+
+    this.create_();
+  }
+
+  /**
+   * Append data to the LCEVC Dec.
+   * @param {BufferSource} data
+   * @param {number} timestampOffset
+   */
+  appendBuffer(data, timestampOffset) {
+    if (this.dec_) {
+      // Timestamp offset describes how much offset should be applied to the
+      // LCEVC enhancement to match the decoded video frame, as such it needs
+      // to be negated.
+      this.dec_.appendBuffer(data, 'video', this.variantId_, -timestampOffset);
+    }
+  }
+
+  /**
+   * Hide the canvas specifically in the case of a DRM Content
+   */
+  hideCanvas() {
+    if (this.dec_) {
+      this.canvas_.classList.add('shaka-hidden');
+    }
+  }
+
+  /**
+   * Create LCEVC Decoder.
+   * @private
+   */
+  create_() {
+    if (this.isSupported_() && !this.dec_) {
+      if (this.lcevcLib_.SupportObject.webGLSupport(this.canvas_)) {
+        // Make sure the canvas is not hidden from a previous playback session.
+        this.canvas_.classList.remove('shaka-hidden');
+        // Initiate LCEVC Dec Library based on the type of module available.
+        if (this.toBeDeprecated_) {
+          this.dec_ = new this.lcevcLib_.LcevcDil(
+              this.media_,
+              this.canvas_,
+              this.decConfig_);
+        } else {
+          this.dec_ = new this.lcevcLib_.LCEVCdec(
+              this.media_,
+              this.canvas_,
+              this.decConfig_);
+        }
+      }
+    }
+  }
+
+  /**
+   * Close LCEVC Decoder.
+   * @override
+   * @export
+   */
+  release() {
+    if (this.dec_) {
+      this.dec_.close();
+      this.dec_ = null;
+    }
+  }
+
+  /**
+   * Check if the LCEVC Decoder lib is present and is supported by the browser.
+   * @return {boolean}
+   * @private
+   */
+  isSupported_() {
+    if (typeof libDPIModule === 'undefined') {
+      shaka.log.alwaysWarn(
+          'Could not find LCEVC Library dependencies on this page');
+    }
+
+    // This is a check to make the LCEVC Dec Integration is backwards compatible
+    // with previous Integration. This logic checks for both implementations
+    // and uses the one available.
+    if (typeof LCEVCdec !== 'undefined') {
+      this.lcevcLib_ = LCEVCdec;
+    } else {
+      if (typeof LcevcDil !== 'undefined') {
+        this.lcevcLib_ = LcevcDil;
+        this.toBeDeprecated_ = true;
+        shaka.log.alwaysWarn(
```

Comment:
Include a `shaka.Deprecate.deprecateFeature`. See: https://github.com/shaka-project/shaka-player/blob/main/lib/player.js#L2099

And you can remove the alwaysWarn

Reply:
Added the `deprecarteFeature` call for the case and removed `alwaysWarn`